wp-content/themes/twentytwenty-child/settings-student.js

```diff
@@ -57,4 +57,23 @@ jQuery(document).ready(function($){
         });
     });
 
-});
\ No newline at end of file
+    var display = Number($(".infinite-scroll").attr("value"));
+    $(window).scroll(function(e){
+        if( $(window).scrollTop() + $(window).height() >= $(document).height() ) {
+            jQuery.ajax({
+                url: my_ajax_object.ajax_url,
+                type: 'POST',
+                data: {
+                    'action' : 'infinite_more_data',
+                    'displayed' : display 
+                }, 
+                success: function(data) {
+                    display++; 
+                    $(".infinite-scroll").append(data);
+                }
+            });
```

Comment:
What if I scroll really fast and trigger multiple requests at the same time, what's gonna happen?

Reply:
It works pretty fine. Could you be more specific of what's wrong?

Reply:
Usually when you have an infinity scroll you should also know when there is a pending request and block all other requests until the pending one resolves. That's done so you don't by mistake request the same data twice and it's just a better user experience.

One more thing, instead of doing the calculations yourselves if your screen is intersecting with a certain object, you can use an [intersection observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). It's super easy to use and does all this stuff for you. No need to change it but I just want you to know for next time.